pkg/provider/loadbalancer_test.go

```diff
@@ -121,376 +125,349 @@ func cmpLoadBalancerStatuses(a, b *corev1.LoadBalancerStatus) bool {
+			loadBalancerIP string
 		)
-		infraServiceExist.Status = corev1.ServiceStatus{
-			LoadBalancer: corev1.LoadBalancerStatus{
-				Ingress: []corev1.LoadBalancerIngress{
-					{
-						IP: loadBalancerIP,
+
+		BeforeAll(func() {
+			ctrl, ctx = gomock.WithContext(context.Background(), GinkgoT())
+			c = mockclient.NewMockClient(ctrl)
+			lb = &loadbalancer{
+				namespace: "test",
+				client:    c,
+				config: LoadBalancerConfig{
+					CreationPollInterval: 1,
+				},
+			}
+
+			tenantService = &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "service1",
+					UID:  types.UID("f6ebf172-2bb1-11e9-b210-d663bd873d93"),
+					Annotations: map[string]string{
+						"annotation-key-1": "annotation-val-1",
+					},
+				},
+				Spec: corev1.ServiceSpec{
+					Type: corev1.ServiceTypeLoadBalancer,
+					Ports: []corev1.ServicePort{
+						{Name: "port1", Protocol: corev1.ProtocolTCP, Port: 80, NodePort: 30001},
+					},
+					ExternalTrafficPolicy: corev1.ServiceExternalTrafficPolicyTypeCluster,
+				},
+			}
+			nodes = []*corev1.Node{}
+			loadBalancerIP = "123.456.7.8"
+
+		})
+
+		It("Should create new Service and poll LoadBalancer service 1 time", func() {
+			checkSvcExistErr := notFoundErr
+			getCount := 1
+			port := 30001
+			infraServiceExist := generateInfraService(
+				tenantService,
+				[]corev1.ServicePort{
+					{Name: "port1", Protocol: corev1.ProtocolTCP, Port: 80, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: int32(port)}},
+				},
+			)
+			infraServiceExist.Status = corev1.ServiceStatus{
+				LoadBalancer: corev1.LoadBalancerStatus{
+					Ingress: []corev1.LoadBalancerIngress{
+						{
+							IP: loadBalancerIP,
+						},
 					},
 				},
-			},
-		}
-		c.EXPECT().Get(
-			ctx,
-			client.ObjectKey{Name: "af6ebf1722bb111e9b210d663bd873d9", Namespace: "test"},
-			gomock.AssignableToTypeOf(&corev1.Service{}),
-		).Do(func(ctx context.Context, key client.ObjectKey, obj client.Object) {
-			if test.checkSvcExistErr == nil {
-				copyService(infraServiceExist, obj.(*corev1.Service))
 			}
-		}).Return(test.checkSvcExistErr)
-		if test.checkSvcExistErr == nil || test.checkSvcExistErr == notFoundErr {
+
+			c.EXPECT().
+				Get(ctx, client.ObjectKey{Name: "af6ebf1722bb111e9b210d663bd873d9", Namespace: "test"}, gomock.AssignableToTypeOf(&corev1.Service{})).
+				Return(checkSvcExistErr)
+
 			infraService1 := generateInfraService(
 				tenantService,
 				[]corev1.ServicePort{
 					{Name: "port1", Protocol: corev1.ProtocolTCP, Port: 80, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: 30001}},
 				},
 			)
-			if test.checkSvcExistErr == nil {
-				infraService1.Status = corev1.ServiceStatus{
-					LoadBalancer: corev1.LoadBalancerStatus{
-						Ingress: []corev1.LoadBalancerIngress{
-							{
-								IP: loadBalancerIP,
+
+			c.EXPECT().Create(ctx, infraService1)
+
+			for i := 0; i < getCount; i++ {
+				infraService2 := &corev1.Service{}
+				copyService(infraService1, infraService2)
```

Comment:
please use `infraService1.DeepCopy()` instead of `copyService` that can be removed.